CMakeLists.txt

```diff
@@ -71,14 +74,73 @@ SET(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
 SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
 
 IF (USE_CUDA)
-    ADD_SUBDIRECTORY(lib/cnmem)
-    LIST(APPEND SINGA_LINKER_LIBS cnmem)
-    SET(global_cuda_objs "")
+    include(ExternalProject)
+    ExternalProject_Add(cnmem
+        GIT_REPOSITORY "https://github.com/NVIDIA/cnmem.git"
```

Comment:
can we just download from a specific commit?
the master could change later, including the APIs.